### ISSUE TITLE
Сontact list (at the left) now has only one visual state (wide-open or icons)

### ISSUE THREAD

**Issue by oleg-indeez** [telegramdesktop/tdesktop#27823]
### Steps to reproduce

Hello
Nothing spcial: just open your Telegram Desktop beta (since 4.16.9)

### Expected behaviour

Since 4.16.9 beta of linux desktop there's small regression (at least I estimate it so). Before that beta I used to have the conacts bar, which is situauted at left side of the app window, to be wide-open in 'top' mode - when no contact selected (see attached no_contact.png), and shrink it to icons when I select any contact (see attached contact.png).

no_contact ![no_contact](https://github.com/telegramdesktop/tdesktop/assets/61268538/43ed1c75-ee4f-4cbe-a5a4-d57a72480384)

contact ![contact](https://github.com/telegramdesktop/tdesktop/assets/61268538/1df157dd-f82c-4ef0-8daa-35e42a797d07)

### Actual behaviour

Now there's single setting for the contacts bar - it's wide-open or icon in both described above states. Please revert it back as I suppose it to be quite informative and useful to see the contact list wide-open (with full-name, last message, its date & status) when I search for the contact and shrink it to icon while chatting.

### Operating system

Linux Mint 21.3 x86_64 (Cinnamon), Kernel: 6.8.7-060807-generic

### Version of Telegram Desktop

Telegram Desktop beta since 4.16.9

### Installation source

Static binary from official website

**Comment by john-preston** [comment]
Yeah, sorry for that. You can see full list if you press space or escape. Previous way didn't let you hiding the names / last messages without an opened chat and it wasn't working so well for people who just wanted to hide them.

**Comment by oleg-indeez** [comment]
Thank you for lightning answer but your suggestion is not what I mean: if I press esc or space it moves me to the search box and shows the channels related - not my contact list. I can post here pics if needed

**Comment by john-preston** [comment]
@oleg-indeez Yeah, sorry, the recent chats and search results are there. Sorry, for now it is what it is.

**Comment by oleg-indeez** [comment]
If you make it open the wide contacts (not search) on pressing esc - that'd be really cool imho. But space (or anything typed when there's no open chat) should obviously go to search box - you're quite right imho

**Comment by oleg-indeez** [comment]
hello. I'm still ask you to consider this proposed change in UI: when one press `esc` being in topmost menu and if the contact list is is compact view - show its wide-view (or may be even swap short <> wide views)
I see that in the latest build (5.0) the topmost-search is activated by `space` key only, `esc` now has no (at least no visible) function at the topmost level

**Comment by dom6770** [comment]
Yes, please change back the behavior! I always press ESC after I'm done with a chat to have the chat list in the "wide" mode where parts of the last message are displayed, right now it's just the user avatars and it's quite annoying.

**Comment by Adam-Kay** [comment]
I'm also asking for this behavior to be reverted, not being able to see the latest message in my recent chats when I press <kbd>Esc</kbd> is very unfortunate.

**Comment by vaaski** [comment]
> it wasn't working so well for people who just wanted to hide them.

I don't think that justifies changing the default behavior that's been there for years without any reasonable option to reverse it.

I would suggest changing it back to how it was, but having <kbd>esc</kbd> toggle between the expanded and collapsed states.

That would not break the muscle memory for anyone that has been using Telegram Desktop for more than 2 weeks and gives the option to completely hide for those that want it.

**Comment by Dregu** [comment]
I also had to downgrade to get this expanding list behavior on esc back, cause I use it all the time to view the names of nondescript group icons etc, but usually have collapsed icons to maximize chat content. A simple toggle option to bring the old behavior back would be fine.

> I would suggest changing it back to how it was, but having esc toggle between the expanded and collapsed states.

I on the other hand like using double esc to return to all chats quickly from a folder, so changing that to a toggle would also need to be optional.

**Comment by oleg-indeez** [comment]
> I on the other hand like using double esc to return to all chats quickly from a folder, so changing that to a toggle would also need to be optional.
well, let it be special advanced setting - it'd be quite logical and convenient (as for myself)

> I guess what we actually want is a new optional hotkey (TAB) to simply toggle the collapsed state of the chat list without closing the current chat. (I know TAB is used for navigation, but it's hardly utilized in this program and I could live without it.)
why should we have a new hotkey for the function that already existed in the client for ages? And, by the way, why the original tag `bug' was altered to `enhancement`? I wrote the report on this case just because some good and useful UI function has gone and I really miss it - it's more like a bug. In all the builds up to (incl.) 4.16.8 anyone could easily setup hist contact list to be either short or wide-open view for both state (open chat / topmost menu w/o any chat open) separately and I really do not understand who (and why) this setting bothered so much that now there's just single view-mode for both states.

**Comment by vaaski** [comment]
I haven't really considered folders and the use of <kbd>esc</kbd> with it. So the easiest solution is probably an advanced setting that toggles between the new and old behavior.

I would even be fine with having the new one as default although the other way around would make more sense to me.

**Comment by dom6770** [comment]
Would a toggle in the settings even be necessary?

Right now with the latest version, I can resize the chat bar, so it's either collapsed or expanded. This settings applies to both views (chat open vs no chat open). Why not simply save two different settings? One when a chat is open, and one when no chat is open, so I resize depending on the state.

So in my case I could collapse the chat list when a chat is open and expand it when no chat is open.

**Comment by Dregu** [comment]
> why should we have a new hotkey for the function that already existed in the client for ages?

The function to toggle the list with one key has never existed afaik. I had the impression that people were using the old functionality to just quickly check the expanded list, but the old functionality is not perfect for this because it is one way and closes the current chat, which you then have to open with a different key if what you wanted was just to quickly toggle it. I also wasn't suggesting the old functionality shouldn't be restored, but was offering a new alternative to do what I actually want to do.

**Comment by vaaski** [comment]
> Why not simply save two different settings? One when a chat is open, and one when no chat is open, so I resize depending on the state.

I could see that actually being great and providing even more flexibility, like having the sidebar expanded when a chat is selected and a collapsed when nothing is selected to either show a lot of information or none at all at the press of a button.

**Comment by oleg-indeez** [comment]
> Would a toggle in the settings even be necessary?

IMHO it's not necessary at all

> Right now with the latest version, I can resize the chat bar, so it's either collapsed or expanded. This settings applies to both views (chat open vs no chat open). Why not simply save two different settings? One when a chat is open, and one when no chat is open, so I resize depending on the state.

and that's just what it was before 4.16.9 and what I used to and eagerly ask to revert to. Contact side list in collapsed view (and alterable by end-user) when any chat is open and wide-open view (also alterable by end-user) if there's no chat opened. Nothing else

**Comment by MayaPosch** [comment]
I concur with @oleg-indeez. This new behaviour is most annoying as I do not want to have the sidebar take up precious space/be distracting while I'm in a chat, while when I'm in the main view having the chat summaries available is very practical. This still worked just fine in version 4.99.

I have opted to disable automatic updates and will be remaining on 4.x until the 5.x versions fix this regression.

**Comment by Kithio** [comment]
I was extremely confused when my telegram client suddenly stopped having the old behavior where the user list was expanded upon closing a chat and only letting it be full "wide" or icons only mode regardless if chat is open or not...

This should be an option for users who prefer having a dynamic list, the sudden change made me get increasingly frustrated over the last few days while looking for answers online... Found no talks about this anywhere which confused me, only to find this issue listing today

I want to see the addition of an OPTION to toggle the old behavior back at the very least if you wish to truly stick to this being what you want as default going fowards... It makes no sense to me that the user list would be collapsed (Icons Only) when no chat is open, it just means you have a large blank space that says "Select a chat to start messaging"

**Comment by oleg-indeez** [comment]
also downgraded to beta 4.16.8

Hope the developer would either revert to the `good-old' UI behaviour or make it option-driven (or two separate setting for contact list appearance when any chat is open or not)  soon

**Comment by john-preston** [comment]
Ok, in the upcoming 5.0.2 version it will work that way:
- first you collapse your chats list while being in the chat, the column becomes userpic-only and you see a lot of chat area
- then you close the chat and it stays narrow, userpic-only column, like it works right now in 5.0.1
- then without any chat open you expand the chats list so you see the content normally there
- after those steps you get the mode you want (it stays expanded when the chat is closed and it stays collapsed userpic-only in case something is opened in the right part)

**Comment by MayaPosch** [comment]
It would be great if there was more of a UX feedback that tells the user what to expect of the sidebar in each mode, but I'll be happy to use the latest version of Telegram Desktop again. Also because I have found that 'disable auto-updates' is very buggy, possibly because a Windows reboot while Telegram is open may trigger the 'forced update' routine as covered in e.g. this ticket: https://github.com/telegramdesktop/tdesktop/issues/26078#issuecomment-1479356243

Regardless, thank you for this update.
